src/main/scala/apps/convolution.scala

```diff
@@ -143,9 +143,7 @@ object convolution {
     matrix: Array[Array[Float]],
     weights: Array[Float]
   ): (Array[Float], TimeSpan[Time.ms]) = {
-    val f = k.as[ScalaFunction `(`
-      Array[Array[Float]] `,` Array[Float]
-      `)=>` Array[Float]]
+    val f = k.as[Args `(` Array[Array[Float]] `,` Array[Float], Array[Float]]
```

Comment:
I find this syntax very confusing, especially with the missing `)`. Can we have something like:
```scala
k.as[`In:` Array[Array[Float]] `,` Array[Float], `Out:` Array[Float]]
```

Reply:
Don't obsess over this, I know already how to fix this in a much nicer way in Scala 3 using [match types](https://docs.scala-lang.org/scala3/reference/new-types/match-types.html):
https://scastie.scala-lang.org/ndV4Xu8RQZWlZRYCc6GAig